File: frontend/lib/models/deepfake_response.dart
class DeepfakeResponse {
  final String timestamp;
  final bool isSynthetic;
  final double confidenceScore;
  final List<String> flags;

  DeepfakeResponse({
    required this.timestamp,
    required this.isSynthetic,
    required this.confidenceScore,
    required this.flags,
  });

  factory DeepfakeResponse.fromJson(Map<String, dynamic> json) {
    final isSynthetic = json['is_synthetic'] as bool? ?? json['is_deepfake'] as bool? ?? false;
    
    double confidence = (json['confidence_score'] ?? 0.0).toDouble();
    if (confidence > 1.0) {
      confidence = confidence / 100.0;
    }

    final List<String> flags = [];
    if (json['flags'] != null) {
      flags.addAll(List<String>.from(json['flags']));
    } else if (json['analysis_details'] != null) {
      final details = json['analysis_details'] as String;
      if (details.contains('|')) {
        flags.addAll(details.split('|').map((s) => s.trim()).where((s) => s.isNotEmpty));
      } else {
        flags.add(details);
      }
    }

    return DeepfakeResponse(
      timestamp: json['timestamp'] as String? ?? DateTime.now().toIso8601String(),
      isSynthetic: isSynthetic,
      confidenceScore: confidence,
      flags: flags,
    );
  }
}

File: frontend/app/src/main/java/com/aegis/scamfirewall/models/DeepfakeResponse.kt
package com.aegis.scamfirewall.models

import com.google.gson.annotations.SerializedName

data class DeepfakeResponse(
    @SerializedName("is_deepfake") val isSynthetic: Boolean,
    @SerializedName("confidence_score") val rawConfidenceScore: Float,
    @SerializedName("analysis_details") val rawDetails: String?
) {
    // Normalised confidence score between 0.0 and 1.0
    val confidenceScore: Float
        get() = if (rawConfidenceScore > 1.0f) rawConfidenceScore / 100.0f else rawConfidenceScore

    val flags: List<String>
        get() = rawDetails?.split("|")?.map { it.trim() }?.filter { it.contains("TTS") || it.contains("synthetic") || it.contains("vocoder") } ?: emptyList()
}
